### PR TITLE
Fixed #14497 -- Improved admin widget for "read only" FileFields

### PR DESCRIPTION
**Pull request by xrmx**
Based on patch by Adam J Forster, Paul Collins and Julien.

### CODE REVIEW

File: django/contrib/admin/utils.py [hunk 1]
```diff
@@ -389,6 +390,8 @@ def display_for_field(value, field):
         return formats.number_format(value, field.decimal_places)
     elif isinstance(field, models.FloatField):
         return formats.number_format(value)
+    elif isinstance(field, models.FileField):
+        return mark_safe('<a href="%s">%s</a>' % (value.url, value))
```
Comment (collinanderson) [inline django/contrib/admin/utils.py]: I'd `conditional_escape()` each argument.

**Comment by xrmx** [comment]
Updated pull request after @collinanderson review suggesting to use conditional_escape on displayed values

**Comment by collinanderson** [comment]
buildbot, test this please

**Comment by xrmx** [comment]
@collinanderson hopefully fixed tests failing because of SimpleTestCase.urls deprecations

**Comment by collinanderson** [comment]
buildbot, test this please

**Comment by collinanderson** [comment]
flake8 warnings:

../tests/admin_widgets/widgetadmin.py:26:1: E302 expected 2 blank lines, found 1
../tests/admin_widgets/widgetadmin.py:30:1: E302 expected 2 blank lines, found 1
../tests/admin_widgets/tests.py:386:120: E201 whitespace after '{'
../tests/admin_widgets/tests.py:386:159: E202 whitespace before '}'

Also, start your commit message like: Fixed #14497 --

**Comment by xrmx** [comment]
Il 03/02/2015 21:29, Collin Anderson ha scritto:
> flake8 warnings:
>
> ../tests/admin_widgets/widgetadmin.py:26:1: E302 expected 2 blank lines,
> found 1
> ../tests/admin_widgets/widgetadmin.py:30:1: E302 expected 2 blank lines,
> found 1
> ../tests/admin_widgets/tests.py:386:120: E201 whitespace after '{'
> ../tests/admin_widgets/tests.py:386:159: E202 whitespace before '}'

Fixed

> Also, start your commit message like: Fixed #14497 --

Done

**Comment by collinanderson** [comment]
buildbot, test this please

**Comment by timgraham** [comment]
merged in 2be621e44c1b5b68c895383f3e20b1f17ddb447a, thanks.

**Pull request closed without merging**